openhands/runtime/base.py

```diff
@@ -216,56 +224,65 @@ def on_event(self, event: Event) -> None:
         if isinstance(event, Action):
             asyncio.get_event_loop().run_until_complete(self._handle_action(event))
 
+    async def _export_latest_git_provider_tokens(self, event: Action) -> None:
+        """
+        Refresh runtime provider tokens when agent attemps to run action with provider token
+        """
+        if not self.git_provider_tokens:
```

Comment:
Also, this function should really only run in SAAS environment (not locally because user PATs don't need to be refreshed, the user controls it)

```suggestion
        if not self.git_provider_tokens or not self.external_auth_id:
```